src/array.c

```diff
@@ -64,7 +64,7 @@ static jl_array_t *_new_array_(jl_value_t *atype, uint32_t ndims, size_t *dims,
         size_t di = dims[i];
         wideint_t prod = (wideint_t)nel * (wideint_t)di;
         if (prod > (wideint_t) MAXINTVAL || di > MAXINTVAL)
-            jl_error("invalid Array dimensions");
+            throw ArgumentError("invalid Array dimensions");
```

Comment:
This file is C code, not Julia code. The changed code here is not valid C.

Reply:
This is also not valid Julia code.